### PR TITLE
DEX-1808 SDK v2

### PR DESCRIPTION
**Pull request by EvansPie**

### CODE REVIEW

File: ios/DataModels/PrimerSettingsRN.swift [hunk 1]
```diff
@@ -1,168 +1,13 @@
...
-    let is3DSOnVaultingEnabled: Bool?
-    let is3DSDevelopmentModeEnabled: Bool?
-    let locale: String?
-    let ios: IosOptionsRN?
-}
-
-fileprivate extension String {
-    var localeFormatted: Locale {
-        return Locale(identifier: self)
+        return PrimerSettings()
```
Comment (d4r1091) [inline ios/DataModels/PrimerSettingsRN.swift:11]: can we double-check whether the whole section can be removed 🙏

File: ios/RNTPrimer.swift [hunk 2]
```diff
@@ -46,13 +48,16 @@ enum PrimerEvents: Int, CaseIterable {
...
-    private var clientTokenCallback: ((String?, Error?) -> Void)?
-    private var isOnClientSessionActionsImplementedInRN: Bool?
-    private var onClientSessionActions: ((String?, Error?) -> Void)?
-    private var onTokenizeSuccess: ((String?, Error?) -> Void)?
-    private var onResumeSuccess: ((String?, Error?) -> Void)?
-    private var implementedReactNativeCallbacks: ImplementedReactNativeCallbacks?
+//    private var isClientTokenCallbackImplementedInRN: Bool?
+//    private var clientTokenCallback: ((String?, Error?) -> Void)?
+//    private var isOnClientSessionActionsImplementedInRN: Bool?
+//    private var onClientSessionActions: ((String?, Error?) -> Void)?
```
Comment (d4r1091) [inline ios/RNTPrimer.swift:54]: if not needed, then 🧹 🧹 🧹  🙏

**Review by d4r1091 — COMMENTED** [review]
Just few things to 🧹 🧹

File: src/models/PrimerClientSession.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+export type PrimerClientSession = IPrimerClientSession;
```
Comment (RockPaperSmittie) [inline src/models/PrimerClientSession.ts:1]: Can we change these exports to only export the interface and then export all interfaces as types in `index.tsx`?

**Review by RockPaperSmittie — APPROVED** [review]
Looking good, just a couple of comments! Will create PR for interface changes and prettier updates

File: src/Primer.ts [hunk 4]
```diff
@@ -1,252 +1,409 @@
...
-import type { PrimerPaymentMethodIntent } from './models/primer-intent';
-import type { PrimerResumeHandler } from './models/primer-request';
-import RNPrimer, { IPrimerError } from './RNPrimer';
+import type { PrimerCheckoutData } from './models/PrimerCheckoutData';
+import type { Primer as IPrimer, PrimerErrorHandler, PrimerPaymentCreationHandler, PrimerResumeHandler, PrimerTokenizationHandler } from './models/Primer';
+import type { PrimerSessionIntent } from './models/PrimerSessionIntent';
+import type { PrimerSettings } from './models/PrimerSettings';
+import RNPrimer from './RNPrimer';
+import type { PrimerCheckoutPaymentMethodData } from './models/PrimerCheckoutPaymentMethodData';
+import type { PrimerClientSession } from 'lib/typescript';
```
Comment (RockPaperSmittie) [inline src/Primer.ts:7]: This doesn't seem right, shouldn't this be imported from a models file?

**Review by RockPaperSmittie — APPROVED** [review]

File: src/models/Primer.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+export interface PrimerResumeHandler {
+  handleSuccess(): void;
+  handleFailure(errorMessage: string | null): void;
+  continueWithNewClientToken(newClientToken: string): void;
+}
+
+export interface PrimerErrorHandler {
+  handleFailure(errorMessage: string | null): void;
```
Comment (semirp) [inline src/models/Primer.ts:34]: This one has a mismatch with native API. On the bridge it is called `showErrorMessage`

File: src/models/PrimerSessionIntent.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export enum PrimerSessionIntent {
```
Comment (semirp) [inline src/models/PrimerSessionIntent.ts:1]: `PrimerPaymentMethodIntent` I am using this one. Should we all align? Not sure what's the best option

File: src/Primer.ts [hunk 7]
```diff
@@ -1,252 +1,409 @@
...
+          onClientSessionUpdate: (primerSettings?.onClientSessionUpdate !== undefined),
+          onTokenizeSuccess: (primerSettings?.onTokenizeSuccess !== undefined),
+          onResumeSuccess: (primerSettings?.onResumeSuccess !== undefined),
+          onDismiss: (primerSettings?.onDismiss !== undefined),
+          onCheckoutFail: (primerSettings?.onCheckoutFail !== undefined),
+        };
+
+        await RNPrimer.setImplementedRNCallbacks(implementedRNCallbacks);
+
+        if (implementedRNCallbacks.primerDidCompleteCheckoutWithData) {
```
Comment (semirp) [inline src/Primer.ts:141]: All of these will fail, as they are checking for wrong callbacks, let's fix that @EvansPie

File: ios/DataModels/PrimerSettingsRN.swift (hunk 1 above)
Comment (EvansPie) [reply ios/DataModels/PrimerSettingsRN.swift]: done

**Pull request closed without merging**